### DISCUSSION TITLE
Airflow: Confusion in execution_date & start_date

### DISCUSSION THREAD

**Discussion by Vishal2696** [apache/airflow#28833] [Q&A]
**Airflow version:** 2.2.3
**Deployment:** Kubernetes
**Executor:** Celery

My airflow scheduled runs were running perfectly until it misbehaved on 31-12-2022. I'm unable to figure out what exactly happened but I'm sure it is related to `exec_date` and `start_date`. My start_date for the dag is determined by the below code.(I call the below function) 
```
def derive_start_date(timezone='Australia/Melbourne') -> datetime:
        def current_year():
            return datetime.datetime.now().year
        local_tz = pendulum.timezone(timezone)
        _dt = datetime.datetime(current_year(), 1, 1, tzinfo=local_tz
        return _dt
```
While the above function should be setting the start date to 01-01-2022 but for some reason in the Airflow UI the dags had the start_date as 31-12-2022 (unable to find why).
The dags run were running fine for months until 31-12-2022. Below are details of a sample run on 30/12/2022. 
```
30/12/22: 
Schedule interval: (30 19 * * *)
Execution Date: 2022-12-30 08:30:00+00:00
DAG Start date: 2022-12-31T13:00:00+00:00
Task start date: 2022-12-31, 8:30
```
```
31/12/2022:
Schedule interval: (30 19 * * *)
Execution Date: 2022-12-31T13:00:00+00:00
DAG Start date: 2022-12-31T13:00:00+00:00
Task start date: 2023-01-01, 12:13
```

```
01/01/2023:
Schedule interval: (30 19 * * *)
Execution Date: 2023-01-01 08:30:00+00:00
DAG Start date: 2022-12-31T13:00:00+00:00
Task start date: 2023-01-02, 8:30
```
As you can see on 31 December the execution dates have changed automatically and I'm unable figure out what could have happened. Can anyone point out which part of the puzzle am I missing here? They are back to normal from 1st January onwards.

**Comment by Taragolis** [comment]
I'm sure that the problem with the function which calculated `start_date` because it could overlap execution interval, as result DAG not started.

If you use cron expressions then Airflow starts DagRuns after `data_interval_end` and set `execution_date` equal to `data_interval_start`. Airflow doesn't start DagRun if `data_interval_start` < `start_date`

Some sample
<img width="934" alt="image" src="https://user-images.githubusercontent.com/3998685/211631735-e702e06d-88a2-4b56-abc6-2b5b87d7b851.png">
